dandi/files/bids.py

```diff
@@ -66,25 +66,29 @@ def _validate(self) -> None:
                 bids_paths = [str(self.filepath)] + [
                     str(asset.filepath) for asset in self.dataset_files
                 ]
+                # TODO gh-943: use RFed data structures, avoid duplicating logic
                 results = validate_bids(*bids_paths)
                 self._dataset_errors: list[str] = []
-                if len(results["path_listing"]) == len(results["path_tracking"]):
-                    self._dataset_errors.append("No valid BIDS files were found")
-                for entry in results["schema_tracking"]:
-                    if entry["mandatory"]:
+                self._asset_errors = defaultdict(list)
+                for i in results:
```

Comment:
please avoid such overly too generic `i` since you inadvertingly would run into overloading it. Make it at least `r` for (`r`esults) or `result`